src/core_plugins/elasticsearch/lib/map_uri.js

```diff
@@ -29,7 +29,7 @@ export default function mapUri(cluster, proxyPrefix) {
     };
 
     // pathname
-    const reqSubPath = request.path.replace(proxyPrefix, '');
+    const reqSubPath = request.path.replace(RegExp("^"+serverPrefix+proxyPrefix), '');
```

Comment:
can we compile the regular expression in `create_proxy.js` and pass it in as an argument?